src/app/services/send-page-data/send-page-data.ts

```diff
@@ -0,0 +1,211 @@
+    'estConvToNav': undefined,
+    'estConvFromNav': undefined,
+    'estNavAfterNavtechFee': undefined,
+    'estTime': undefined,
+    'changellyFeeTotalToNav': undefined,
+    'navtechFeeTotal': undefined,
+    'changellyFeeTotalFromNav': undefined,
+    'minTransferAmount': undefined,
+    'validData': false,
+    'errors': {
+      'invalidDestAddress': false,
+      'invalidTransferAmount': false,
+      'transferTooSmall': false,
+      'transferTooLarge': false,
+      'navToNavTransfer': false,
+      'changellyError': false,
+    }
+  }
+
+  CHANGELLY_FEE: number = changellyConstData.CHANGELLY_FEE
+  NAVTECH_FEE: number = changellyConstData.NAVTECH_FEE
+  MAX_NAV_PER_TRADE: number = changellyConstData.MAX_NAV_PER_TRADE
+
+  dataSubject = new Subject<any>()
+
+  dataSetSubject = new Subject<any>()
+
+  isDataSet: boolean = false
+
+  constructor(private changellyApi: ChangellyApiService) { }
+
+  getData(): void {
+    if(this.dataStored){
+      this.dataSubject.next(this.dataBundle)
+    }
+  }
+
+  getDataStream(): Observable<any> {
+    return this.dataSubject.asObservable()
+  }
+
+  getDataStatusStream(): Observable<any> {
+    return this.dataSetSubject.asObservable()
+  }
+
+  clearData(): void {
+    this.dataBundle =  {
+      'transferAmount': undefined,
+      'originCoin': undefined,
+      'destCoin': undefined,
+      'destAddr': undefined,
+      'estConvToNav': undefined,
+      'estConvFromNav': undefined,
+      'estNavAfterNavtechFee': undefined,
+      'estTime': undefined,
+      'changellyFeeTotalToNav': undefined,
+      'navtechFeeTotal': undefined,
+      'changellyFeeTotalFromNav': undefined,
+      'minTransferAmount': undefined,
+      'validData': false,
+      'errors': {
+        'invalidDestAddress': false,
+        'invalidTransferAmount': false,
+        'transferTooSmall': false,
+        'transferTooLarge': false,
+        'navToNavTransfer': false,
+        'changellyError': false,
+      }
+    }
+    this.dataSubject.next(this.dataBundle)
+  }
+
+  checkIsDataSet():boolean {
+    return this.isDataSet
+  }
+
+  storeData(transferAmount, originCoin, destCoin, destAddr): void {
+    this.dataSetSubject.next(false)
+    this.resetDataBundleErrors(this.dataBundle)
+    this.dataStored = false
+    this.dataBundle.transferAmount = Number(transferAmount) ? Number(transferAmount): undefined
+    this.dataBundle.originCoin = originCoin
+    this.dataBundle.destCoin = destCoin
+    this.dataBundle.destAddr = destAddr
+    this.dataStored = true
+    if(!this.validateFormData(this.dataBundle)) {
+      this.dataSubject.next(this.dataBundle)
+      return //validation errors, so return early
+    }
+
+    this.getEstimatedExchange(originCoin, 'nav', transferAmount)
+      .then((data) => {
+        if(originCoin === 'nav'){
+          this.dataBundle.changellyFeeTotalToNav = 0
+        } else {
+          this.dataBundle.changellyFeeTotalToNav = transferAmount - transferAmount * this.CHANGELLY_FEE
```

Comment:
I'd try to avoid nesting your promises like this, you'll probably find its hard to test. Try breaking them out, even if its just into multiple functions which get called from inside the previous promise